src/cmd/add.nim

```diff
@@ -5,13 +5,32 @@ import ../cli/term, ../cli/prompt
 import ../util/flow
 import ../modpack/files, ../modpack/install
 
+proc addDependencies(manifest: var Manifest, file: ManifestFile,
+    strategy: string): void =
+  ## Recursively add dependencies of a mod
+  if len(file.dependencies) > 0:
+    for id in file.dependencies:
+      let cfMod = waitFor(fetchMod(id))
+      if manifest.isInstalled(id):
+        continue
+      let cfModFiles = waitFor(fetchModFiles(id))
+      let selectedCfModFile = cfModFiles.selectModFile(manifest, strategy)
+      if selectedCfModFile.isNone:
+        echoError "Warning: unable to resolve dependencies."
+      let cfModFile = selectedCfModFile.get()
+      echoInfo "Installing ", cfMod.name.cyanFg, ".."
+      let modToInstall = initManifestFile(id, cfModFile.fileId, cfMod.name,
```

Comment:
Since it's 5 properties now and Nim has [named arguments](https://nim-lang.org/docs/tut1.html#procedures-named-arguments) you could do

```nim
... = initManifestFile(
  projectId = id,
  fileId = cfModFile.fileId,
  name = cfMod.name,
  explicit = false,
  dependencies = cfModFile.dependencies
);
```

which is a bit more text, but much more readable (especially with boolean arguments).